src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3547,6 +3553,20 @@ private BoundNode VisitLValue(BoundNode node)
             return Visit(node, expressionIsRead: false);
         }
 
+        private bool TypeAllowsConditionalState(TypeSymbol? type)
+        {
+            return type is not null
+                && (type.SpecialType == SpecialType.System_Boolean || type.IsDynamic() || type.IsErrorType());
```

Comment:
:memo: We could disallow conditional state for `dynamic` but that degrades a couple of tests.

Reply:
:memo: The error type scenario is hit with `LiftedRelationalOperation_UserDefined`. Presumably, we could move the fix upstream (unsplit some time during binary operator analysis). Let me know if you have a preference

Reply:
I think it's better to permit error type here than to try and adjust the bound tree shape.